packages/autoloader/src/AutoloadGenerator.php

```diff
@@ -0,0 +1,372 @@
+		// } else {
+		// $filesCode .= '     require ' . $this->getPathCode($filesystem, $basePath, $vendorPath,
+		// $functionFile) . ";\n";
+		// }
+		// }
+
+		$classmapFile = <<<EOF
+<?php
+
+// autoload_classmap_package.php @generated by automattic/autoload
+
+\$vendorDir = $vendorPathCode;
+\$baseDir = $appBaseDirCode;
+
+return array(
+
+EOF;
+
+		$blacklist = null;
+		if ( ! empty( $autoloads['exclude-from-classmap'] ) ) {
+			$blacklist = '{(' . implode( '|', $autoloads['exclude-from-classmap'] ) . ')}';
+		}
+
+		// flatten array
+		$classMap = array();
+		if ( $scanPsr0Packages ) {
+			$namespacesToScan = array();
+
+			// Scan the PSR-0/4 directories for class files, and add them to the class map
+			foreach ( array( 'psr-0', 'psr-4' ) as $psrType ) {
+				foreach ( $autoloads[ $psrType ] as $namespace => $info ) {
+					$version = array_reduce(
+						array_map(
+							function( $item ) {
+									return $item['version']; },
+							$info
+						),
+						function( $carry, $version ) {
+							return version_compare( $version, $carry, '>' ) ? $version : $carry;
+						},
+						0
+					);
+
+					$namespacesToScan[ $namespace ][] = array(
+						'paths'   => array_map(
+							function( $item ) {
+									return $item['path']; },
+							$info
+						),
+						'type'    => $psrType,
+						'name'    => array_map(
+							function( $item ) {
+									return $item['name']; },
+							$info
+						),
+						'version' => $version,
+					);
+				}
+			}
+
+			krsort( $namespacesToScan );
+
+			foreach ( $namespacesToScan as $namespace => $groups ) {
+				foreach ( $groups as $group ) {
+					$psrType = $group['type'];
+					foreach ( $group['paths'] as $dir ) {
+						$dir = $filesystem->normalizePath( $filesystem->isAbsolutePath( $dir ) ? $dir : $basePath . '/' . $dir );
+						if ( ! is_dir( $dir ) ) {
+							continue;
+						}
+
+						$namespaceFilter = $namespace === '' ? null : $namespace;
+						$classMap        = $this->addClassMapCode(
+							$filesystem,
+							$basePath,
+							$vendorPath,
+							$dir,
+							$blacklist,
+							$namespaceFilter,
+							$group['version'],
+							$classMap
+						);
+					}
+				}
+			}
+		}
+		// todo: test this out.
+		// foreach ($autoloads['classmap'] as $dir) {
+		// $classMap = $this->addClassMapCode($filesystem, $basePath, $vendorPath, $dir, $blacklist, null, '1', $classMap);
+		// }
+
+		ksort( $classMap );
+		foreach ( $classMap as $class => $code ) {
+			$classmapFile .= '    ' . var_export( $class, true ) . ' => ' . $code;
+		}
+		$classmapFile .= ");\n";
+
+		file_put_contents( $targetDir . '/autoload_classmap_package.php', $classmapFile );
+
+		$sourceLoader = fopen( __DIR__ . '/autoload.php', 'r' );
+		$targetLoader = fopen( $vendorPath . '/autoload_packages.php', 'w+' );
+		fwrite( $targetLoader, stream_get_contents( $sourceLoader ) );
+
+	}
+
+	/**
+	 * This function differes from the composer parseAutoloadsType in that beside returning the path.
```

Comment:
Not sure I can understand the comment here.

```suggestion
	 * This function differs from the composer parseAutoloadsType in that beside returning the path.
```